.gitlab-ci.yml

```diff
@@ -1,102 +1,101 @@
 stages:
     - build
-    - deploy
+    - deploy-preview
+    - deploy-prod
 
 variables:
-    CONTAINER_IMAGE: ${CI_REGISTRY_IMAGE}:${CI_COMMIT_SHORT_SHA}
+    DOCKER_IMAGE: ${CI_REGISTRY_IMAGE}:${CI_COMMIT_SHORT_SHA}
     DOCKER_TLS_CERTDIR: ""
 
 .branch-rules-main:
     rules:
-        - if: '$CI_DEPLOY_FREEZE == null && $CI_COMMIT_BRANCH == "main"'
+        - if: $CI_DEPLOY_FREEZE == null && $CI_COMMIT_BRANCH == "main"
           when: on_success
-        - when: never
 
-.branch-rules-dev:
+.branch-rules-review:
+    extends: .branch-rules-main
     rules:
-        - if: '$CI_DEPLOY_FREEZE == null && $CI_COMMIT_BRANCH == "dev"'
-          when: on_success
-        - when: never
-
-.build-rules:
-    rules:
-        - if: '$CI_DEPLOY_FREEZE == null'
-          when: on_success
-        - when: never
+        - if: $CI_DEPLOY_FREEZE == null && $CI_COMMIT_BRANCH != "main"
+          when: manual
 
 #Base templates
 .deploy:
-    stage: deploy
     image: dockerhub.ebi.ac.uk/ensembl-web/deploy-tools:latest
     before_script:
         - kubectl config use-context ${AGENT}
         - kubectl config set-context --current --namespace=${NS}
     script:
         - git clone --depth 1 --branch k8s123-migration https://gitlab.ebi.ac.uk/ensembl-web/ensembl-k8s-manifests.git
         - cd ensembl-k8s-manifests/ensembl-web-track-api
-        - kustomize edit set image DOCKER_IMAGE=${CONTAINER_IMAGE}
+        - kustomize edit set image DOCKER_IMAGE=${DOCKER_IMAGE}
         - kubectl apply -k .
 
 .deploy-main:
     extends:
         - .deploy
         - .branch-rules-main
 
-.deploy-dev:
+.deploy-all:
```

Comment:
"all" meaning "other than main"? I got confused here for a minute, thinking that it means "deploy everything". Perhaps `.deploy-review` or `.deploy-non-main` would better reflect the purpose of this template?

Reply:
Renamed